ee/debug/checkups/util.go

```diff
@@ -107,6 +116,47 @@ func addFileToZip(z *zip.Writer, location string) error {
 	return nil
 }
 
+func addStreamToZip(z *zip.Writer, name string, modTime time.Time, reader io.Reader) error {
+	// Create metadata file first
+	metaout, err := z.Create(name + ".flaremeta")
+	if err != nil {
+		return fmt.Errorf("creating %s in zip: %w", name+".flaremeta", err)
+	}
+
+	// Marshal metadata
+	b, err := json.Marshal(fileInfo{
+		Name:    filepath.Base(name),
+		ModTime: modTime,
+	})
+	if err != nil {
+		return fmt.Errorf("marshalling json: %w", err)
+	}
+
+	var buf bytes.Buffer
+	if err := json.Indent(&buf, b, "", "  "); err != nil {
+		return fmt.Errorf("indenting json: %w", err)
+	}
+	metaout.Write(buf.Bytes())
```

Comment:
Would you add a check for the error return on `metaout.Write` here? (I know this is identical to the `Write` call on 86 where we don't check the return argument either -- we should have one there too 😅 .)